src/index.ts

```diff
@@ -135,14 +137,15 @@ export function resolve(
   found: boolean
   path?: string | null
 } {
-  if (!cachedOptions || cachedOptions !== options) {
```

Comment:
This was never going to be true since `cachedOptions` is created from `options` but is not `options` itself.

Reply:
Hah, seems right, I didn't know why I wrote such codes here. 😂 But it can be changed in a new PR if we decide to use something like `hashobject`.

Reply:
Hah, seems right, I didn't know why I wrote such codes here. 😂 But it can be changed in a new PR if we decide to use something like `hashobject`.